src/olympia/blocklist/mlbf.py

```diff
@@ -25,28 +25,33 @@ def get_blocked_guids():
             block.min_version == Block.MIN and
             block.max_version == Block.MAX)
         versions = {
-            version_id: (block.guid, version)
+            version_id: {'guid': block.guid, 'version': version}
             for version, (version_id, _) in block.addon_versions.items()
             if is_all_versions or block.is_version_blocked(version)}
         all_versions.update(versions)
-    return all_versions.values()
+    # and get the hashes from the File records
+    file_hashes = File.objects.filter(
+        version_id__in=all_versions.keys()).values_list('version_id', 'hash')
+    return [
+        (all_versions[v_id]['guid'], all_versions[v_id]['version'], hash)
+        for v_id, hash in file_hashes]
 
 
 def get_all_guids():
-    from olympia.versions.models import Version
+    from olympia.files.models import File
 
-    return Version.unfiltered.values_list('addon__guid', 'version')
+    return File.objects.values_list(
+        'version__addon__guid', 'version__version', 'hash')
 
 
 def hash_filter_inputs(input_list, key_format):
     return [
-        key_format.format(guid=guid, version=version)
-        for (guid, version) in input_list]
+        key_format.format(guid=guid, version=version, xpi_hash=hash)
```

Comment:
Given this line, I'd expect `hash` to be `xpi_hash` everywhere in this patch. That would make it slightly more future-proof as the `key_format` expects `xpi_hash` so potentially a further iteration might introduce `hash` a placeholder (maybe not but we never know and that would still be more consistent)

Reply:
The property is `File.hash` and, imo, just `hash` is ambiguous for Firefox so I went with `xpi_hash`.  So the mismatch is going to happen somewhere.

Reply:
I see. Would it be worth changing here, though?

Reply:
which change are you suggesting exactly?

Reply:
Sorry, renaming `hash` to `xpi_hash` in this file (even if the `File` property is named `hash`).